### PR TITLE
chore: coordinate superset-ui unittest with main repository

### PR DESCRIPTION
**Pull request by zhaoyongjie**
🏠 Internal
coordinate superset-ui unittest with main repository
1. Jest use source code for module import
2. remove dependence: `@airbnb/config-jest`
3. remove `@airbnb/config-jest` related "setup" before unit test
4. refactor some unittest
5. clean up

### CODE REVIEW

**Review by villebro — COMMENTED** [review]
LGTM, but I'd prefer to get @kgabryje @geido or @michael-s-molina to also take a look, as this is unfamiliar territory to me.

File: packages/superset-ui-core/test/translation/Translator.test.ts [hunk 1]
```diff
@@ -34,6 +35,18 @@ configure({
 });
 
 describe('Translator', () => {
+  const spy = jest.spyOn(logging, 'warn');
+
+  beforeAll(() => {
+    spy.mockImplementation((info: any) => {
```
Comment (kgabryje) [inline packages/superset-ui-core/test/translation/Translator.test.ts:41]: Can you explain why do we need to mock logging methods? And why do they throw errors?

File: packages/superset-ui-core/test/utils/logging.test.ts [hunk 2]
```diff
@@ -36,9 +40,16 @@ describe('logging', () => {
     expect(() => {
       logging.error('error');
     }).toThrow('error');
+
+    // to support: npx jest --silent
+    const spy = jest.spyOn(logging, 'trace');
+    spy.mockImplementation(() => {
```
Comment (kgabryje) [inline packages/superset-ui-core/test/utils/logging.test.ts:46]: Should we include `logging.trace`'s argument here? Like:
`throw new Error('Trace: ' + traceMessage)`

Comment (zhaoyongjie) [reply packages/superset-ui-core/test/utils/logging.test.ts:46]: This mock just for `logging.trace()`, Ensure that when call the `logging.trace()`, there is `Trace:` at the beginning of the return.

File: packages/superset-ui-core/test/translation/Translator.test.ts (hunk 1 above)
Comment (kgabryje) [reply packages/superset-ui-core/test/translation/Translator.test.ts:41]: Oh, I see, so we just bring back relevant changes from removed library 👍

Comment (zhaoyongjie) [reply packages/superset-ui-core/test/translation/Translator.test.ts:41]: There are some test cases that need to get warning message.

**Review by kgabryje — APPROVED** [review]
Looks good, but just like Ville I'm not 100% familiar with those configs. Let's have third pair of eyes take a look just to be sure

**Review by ktmud — APPROVED** [review]
lgtm

**Pull request merged**